usaspending_api/transactions/delta_models/transaction_search_dataframe.py

```diff
@@ -0,0 +1,914 @@
+        tfpds.sole_proprietorship,
+        tfpds.solicitation_date,
+        tfpds.solicitation_identifier,
+        tfpds.solicitation_procedur_desc,
+        tfpds.solicitation_procedures,
+        tfpds.state_controlled_instituti,
+        tfpds.subchapter_s_corporation,
+        tfpds.subcontinent_asian_asian_i,
+        tfpds.subcontracting_plan,
+        tfpds.subcontracting_plan_desc,
+        tfpds.the_ability_one_program,
+        tfpds.total_obligated_amount,
+        tfpds.township_local_government,
+        tfpds.transaction_number,
+        tfpds.transit_authority,
+        tfpds.tribal_college,
+        tfpds.tribally_owned_business,
+        tfpds.type_of_contract_pricing,
+        tfpds.type_of_contract_pric_desc,
+        tfpds.type_of_idc,
+        tfpds.type_of_idc_description,
+        tfpds.type_set_aside,
+        tfpds.type_set_aside_description,
+        tfpds.undefinitized_action,
+        tfpds.undefinitized_action_desc,
+        tfpds.us_federal_government,
+        tfpds.us_government_entity,
+        tfpds.us_local_government,
+        tfpds.us_state_government,
+        tfpds.us_tribal_government,
+        tfpds.vendor_doing_as_business_n,
+        tfpds.vendor_fax_number,
+        tfpds.vendor_phone_number,
+        tfpds.veteran_owned_business,
+        tfpds.veterinary_college,
+        tfpds.veterinary_hospital,
+        tfpds.woman_owned_business,
+        tfpds.women_owned_small_business,
+        fed_and_tres_acct.program_activities.cast(StringType()),
+    ]
+
+    df = (
+        tn.join(tfabs, (tn.id == tfabs.transaction_id) & ~tn.is_fpds, "leftouter")
+        .join(tfpds, (tn.id == tfpds.transaction_id) & tn.is_fpds, "leftouter")
+        .join(rcfda, tfabs.cfda_number == rcfda.program_number, "leftouter")
+        .join(rlu, rlu.recipient_hash == recipient_hash, "leftouter")
+        .join(awards, tn.award_id == awards.id, "leftouter")
+        .join(aa, tn.awarding_agency_id == aa.id, "leftouter")
+        .join(taa, aa.toptier_agency_id == taa.toptier_agency_id, "leftouter")
+        .join(saa, aa.subtier_agency_id == saa.subtier_agency_id, "leftouter")
+        .join(aa_id, ((aa_id.toptier_agency_id == taa.toptier_agency_id) & aa_id.toptier_flag), "leftouter")
+        .join(fa, tn.funding_agency_id == fa.id, "leftouter")
+        .join(tfa, fa.funding_toptier_agency_id == tfa.toptier_agency_id, "leftouter")
+        .join(sfa, fa.funding_subtier_agency_id == sfa.subtier_agency_id, "leftouter")
+        .join(fa_id, ((fa_id.toptier_agency_id == tfa.funding_toptier_agency_id) & (fa_id.row_num == 1)), "leftouter")
+        .join(prl, prl.parent_recipient_hash == parent_recipient_hash, "leftouter")
+        .join(
+            recipient_hash_and_levels,
+            (sf.col("recipient_hash") == recipient_hash_and_levels.recipient_level_hash)
+            & ~(
+                sf.col("legal_business_name").isin(
+                    [
+                        "MULTIPLE RECIPIENTS",
```

Comment:
Good case to use this lookup instead now that we have the dataframe approach: https://github.com/fedspendingtransparency/usaspending-api/blob/cf57d470ea1e55916f5b064c78e1949aa36e7b37/usaspending_api/recipient/v2/lookups.py#L17